pkg/controllers/reconciler_test.go

```diff
@@ -61,11 +67,24 @@ func TestUpgrade(test *testing.T) {
 			expectedInitimage: initimage,
 			upgrade:           false,
 		},
+		{
+			envVars: map[string]string{strings.ReplaceAll(strings.ToUpper(filepath.Base(image)), "-", "_") + "_SHA": image + "@sha256:" + sha256,
+				strings.ReplaceAll(strings.ToUpper(filepath.Base(initimage)), "-", "_") + "_SHA": initimage + "@sha256:" + sha256},
+			image:             image + "@sha256:" + preSha256,
+			expectedImage:     image + "@sha256:" + sha256,
+			initimage:         initimage + "@sha256:" + preSha256,
+			expectedInitimage: initimage + "@sha256:" + sha256,
+			upgrade:           true,
+		},
 	}
 
 	for i := range tests {
 		t := tests[i]
 
+		for key, value := range t.envVars {
+			os.Setenv(key, value)
+		}
```

Comment:
Env variables stay from test run to test run. I think we should somehow clear the variables after the run. If more tests are added which do not use the env variables, they could get false-negative results.

Reply:
Thanks! Added `os.clearEnv()`!